cli/command_snapshot_restore.go

```diff
@@ -0,0 +1,36 @@
+package cli
+
+import (
+	"context"
+
+	"github.com/kopia/kopia/fs/localfs"
+	"github.com/kopia/kopia/repo"
+	"github.com/kopia/kopia/repo/manifest"
+	"github.com/kopia/kopia/snapshot"
+	"github.com/kopia/kopia/snapshot/snapshotfs"
+)
+
+var (
+	snapshotRestoreCommand    = snapshotCommands.Command("restore", "Restore a snapshot from the snapshot ID to the given target path")
+	snapshotRestoreSnapID     = snapshotRestoreCommand.Arg("id", "Snapshot ID to be restored").Required().String()
+	snapshotRestoreTargetPath = snapshotRestoreCommand.Arg("target-path", "Path of the directory for the contents to be restored").Required().String()
+)
+
+func runSnapRestoreCommand(ctx context.Context, rep *repo.Repository) error {
```

Comment:
Rename the existing `snapshotfs.Restore` function (and its call points) to `snapshotfs.RestoreRoot`.

Move this function to `snapshotfs/restore.go` as `Restore(ctx context.Context, rep *repo.Repository, snapID manifest.ID) error`

And then see comments below.